src/nvim/ex_docmd.c

```diff
@@ -2204,7 +2205,7 @@ find_ucmd (
             eap->cmdidx = CMD_USER;
           else
             eap->cmdidx = CMD_USER_BUF;
-          eap->argt = (long)uc->uc_argt;
+          eap->argt = uc->uc_argt;
```

Comment:
Note that this all will disappear once eap->argt is refactored into uint32_t too (if that's possible, as I presume).